UMN-AutoPackager/public/Build-MEMCMPackage.ps1

```diff
@@ -0,0 +1,323 @@
+function Build-MEMCMPackage {
+    <#
+    .SYNOPSIS
+    Create an application in Configuration Manager based on the properties of the UMNAutopackager json files.
+    .DESCRIPTION
+    This command creates an application for each site based on the values of the GlobalConfig and PackageConfig json values. It leverages various powershell commands provided with ConfigMgr.
+    .PARAMETER GlobalConfig
+    Input the global configuration json file using the Get-GlobalConfig command
+    .PARAMETER PackageDefinition
+    Input the package definition json file using the Get-GlobalConfig command
+    .PARAMETER Credential
+    Input the credentials object or the user name which will prompt for credentials. If not called will attempt to use the credentials of the account that is running the script.
+    .EXAMPLE
+    Build-MEMCMPackage -GlobalConfig (Get-UMNGlobalConfig -Path C:\UMNAutopackager\GlobalConfig.json) -PackageDefinition (Get-UMNGlobalConfig -Path C:\UMNAutopackager\PackageConfig.json) -Credential MyUserName
+    Runs the function prompting for the credentials of MyUserName.
+    .EXAMPLE
+    Build-MEMCMPackage -GlobalConfig $globaljson -PackageDefinition $pkgjson -Credential $creds
+    Runs the function using the credentials stored in the $creds variable.
+    #>
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Input the values of the GlobalConfig.json.")]
+        [psobject]$GlobalConfig,
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Input the values of the various packagedefinition.json files.")]
+        [psobject[]]$PackageDefinition,
+        [ValidateNotNull()]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty
+    )
+    begin {
+        Write-Information -MessageData "Command = $($myinvocation.mycommand)" -Tags Meta
```

Comment:
If I understand Write-Information it drops data into the pipeline.  Do we foresee any issues writing this info into the pipeline?  I'm not really sure, probably not but just wanted to check.

Reply:
Based on the information in the book Powershell scripting toolmaking, using Write-Information isn’t any different than using Write-Verbose. By default it doesn't produce any output unless you call the -InformationAction Continue. Then it outputs similar to -verbose. It allows tagging and you can output the data to a variable using -InformationVariable nameofvariable.  I thought I would give it a try and used some examples from the book in the begin. Figuring that might be useful for troubleshooting in the future. I'd be fine if we didn't include it.